### DISCUSSION TITLE
How to access shadow element with Driver()

### DISCUSSION THREAD

**Discussion by FuadMannan** [seleniumbase/SeleniumBase#2503] [Q&A] [answered]
Hello,

I'm trying to access an iframe within a shadow element:
![image](https://github.com/seleniumbase/SeleniumBase/assets/28613277/b63eeae5-bfb5-46cc-b390-64aa9c7aed81)

I'm aware that SeleniumBase has a `::shadow` selector, but I'm wondering if this can be used with the Driver() returnable object?
(i.e. `from seleniumbase import Driver` and `driver = Driver(uc=True)`)

I tried `iframe = driver.find_element('div#px-captcha::shadow iframe')` and other variations where I specify attributes used by the iframe, but they all gave an "Invalid or illegal selector was specified" error.

I've seen the [shadow_root_test.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/shadow_root_test.py) and [test_shadow_dom.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_shadow_dom.py) examples, but those are using BaseCase.

On another note, I'm also wondering if the methods listed at https://seleniumbase.io/help_docs/method_summary/ are supposed to be able to be used by the Driver() object? I've tried some of them with the Driver() object but some of them don't work for me. I'm assuming that those ones aren't meant for the Driver() object and that I'm misunderstanding when or why the Driver() object should be used.

My apologies if this seems trivial, I'm still trying to understand how to effectively use SeleniumBase.

**Comment by mdmintz** [comment, marked as answer]
The `::shadow` selector only works on `open` shadow-root elements. From your screenshot, it looks like you're dealing with a `closed` shadow-root element, which isn't reachable. Info: https://github.com/SeleniumHQ/selenium/issues/5869#issuecomment-388818416

The [method_summary](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/method_summary.md) methods are mainly just for `BaseCase` and `SB()` formats. If you scroll to the bottom of that page, you'll find a section for `Driver()` methods:

```python
# "driver"-specific methods added by SeleniumBase`.
```

**Reply by FuadMannan** [reply]
Thank you very much for your incredibly fast response, and for your efforts with this project!
